integration/README.md

```diff
@@ -3,45 +3,70 @@
 
 Integration tests for Cook scheduler
 
-To run the tests on a local cook install, run
+## Running in docker
+
+If you're running cook in docker, it can be beneficial to run the integration tests in a docker container so that the IP addresses resolve correctly. To do so, there are two separate scripts:
 
 ```bash
-$ python setup.py nosetests
+$ bin/build-docker-image.sh
 ```
 
-To configure the Cook scheduler URL, set the `COOK_SCHEDULER_URL` environment variable. The default value is `http://localhost:12321`.
+This needs to be run once, and can be run again in order to include any new dependencies in the docker image.
 
-If you want to run a single test and see the log messages as they occur, you can run, for example:
+```bash
+$ bin/run-integration.sh COOK_PORT_1 COOK_PORT_2
+```
+
+This launches the integration tests in the docker container. You can optionally pass one (or two, if using `COOK_MULTI_CLUSTER`) ports that cook is listening on. This mounts the local directory in the docker container so there is no need to rebuild when changing the tests.
+
+## Running the tests locally
+
+Running the integration tests from your local machine may simplify your development workflow.
+
+### Installing dependencies
+
+Our integration tests currently require Python 3.6.
+All dependencies are specified in [`requirements_dev.txt`](./requirements_dev.txt),
+and can be installed automatically via `pip`:
 
 ```bash
-$ nosetests tests.cook.test_basic:CookTest.test_max_runtime_exceeded --nologcapture
+$ pip install -r requirements.txt
 ```
 
-## Multi-scheduler tests
+### Basic tests
 
-The [test_multi_cook.py](tests/cook/test_multi_cook.py) file contains integration tests that rely on two separate running schedulers. These tests are skipped by default, since we often only want to target a single cook cluster. If you want to include these tests, make sure to set the `COOK_MULTI_CLUSTER` environment variable, for example:
- 
- ```bash
- $ COOK_MULTI_CLUSTER= nosetests tests.cook.test_multi_cook
- ```
+To run the tests on a local cook install, run
 
-To configure the second cook scheduler URL, set the `COOK_SCHEDULER_URL_2` environment variable. The default value is `http://localhost:22321`.
+```bash
+$ pytest
+```
 
-## Running in docker
+To configure the Cook scheduler URL, set the `COOK_SCHEDULER_URL` environment variable. The default value is `http://localhost:12321`.
 
-If you're running cook in docker, it can be beneficial to run the integration tests in a docker container so that the IP addresses resolve correctly. To do so, there are two separate scripts:
+If you want to run a single test and see the log messages as they occur, you can run, for example:
 
 ```bash
-$ bin/build-docker-image.sh
+$ pytest -n 0 --capture=no tests/cook/test_basic.py::CookTest::test_basic_submit
```

Comment:
Should we mention the `single-test.sh` script?

Reply:
That got deleted too! We have a `only-run` script instead. (It lets you specify one or more tests, hence the name change.)

I wasn't sure if I wanted to mention that script in our documentation, but that's probably more user-friendly than suggesting people use the quirky `pytest -k` method directly.